src/lib/slice.zig

```diff
@@ -412,6 +443,89 @@ pub const Slice = struct {
+        const commit_oid = self.ref.target() orelse return SparseError.UNABLE_TO_GET_REFS;
+        
+        const signature = LibGit.GitSignature.default(self.repo) catch |err| {
+            log.err("Failed to create signature for note: {}", .{err});
+            return err;
+        };
+        defer signature.free();
+        
+        // Format note content
+        const note_content = try std.fmt.allocPrint(alloc, "{s}{s}", .{ SLICE_PARENT_NOTE_PREFIX, parent_branch });
+        defer alloc.free(note_content);
+        
+        _ = LibGit.GitNote.create(self.repo, commit_oid, note_content, signature) catch |err| {
+            log.err("Failed to create parent note for slice {s}: {}", .{ self.name(), err });
+            return err;
+        };
+        
+        log.info("Created parent note for slice {s} with parent: {s}", .{ self.name(), parent_branch });
+    }
+
+    /// Reads the parent branch from git notes for this slice
+    pub fn getParentFromNotes(self: *const Slice, alloc: Allocator) !?[]const u8 {
+        const commit_oid = self.ref.target() orelse return null;
+        
+        const note = LibGit.GitNote.read(self.repo, commit_oid) catch |err| {
+            log.debug("Failed to read note for slice {s}: {}", .{ self.name(), err });
+            return null;
+        };
+        
+        if (note == null) {
+            return null; // No note exists
+        }
+        
+        defer note.?.free();
+        
+        const note_content = note.?.content();
+        return parseSliceParentNote(note_content, alloc);
+    }
+
+    /// Updates the parent note for this slice
+    pub fn updateParentNote(self: *Slice, new_parent_branch: []const u8, alloc: Allocator) !void {
+        const commit_oid = self.ref.target() orelse return SparseError.UNABLE_TO_GET_REFS;
+        
+        const signature = LibGit.GitSignature.default(self.repo) catch |err| {
+            log.err("Failed to create signature for note update: {}", .{err});
+            return err;
+        };
+        defer signature.free();
+        
+        // Format note content
+        const note_content = try std.fmt.allocPrint(alloc, "{s}{s}", .{ SLICE_PARENT_NOTE_PREFIX, new_parent_branch });
+        defer alloc.free(note_content);
+        
+        _ = LibGit.GitNote.update(self.repo, commit_oid, note_content, signature) catch |err| {
+            log.err("Failed to update parent note for slice {s}: {}", .{ self.name(), err });
+            return err;
+        };
+        
+        log.info("Updated parent note for slice {s} with new parent: {s}", .{ self.name(), new_parent_branch });
+    }
+
+    /// Removes the parent note for this slice
+    pub fn removeParentNote(self: *Slice) !void {
+        const commit_oid = self.ref.target() orelse return SparseError.UNABLE_TO_GET_REFS;
+        
+        const signature = LibGit.GitSignature.default(self.repo) catch |err| {
```

Comment:
Instead of using `LibGit.GitSignature` import `GitSignature` below in the file and use it.